src/Magento/FunctionalTestingFramework/Test/Objects/ActionObject.php

```diff
@@ -599,6 +601,26 @@ private function validateUrlAreaAgainstActionType($obj)
         }
     }
 
+    /**
+     * Validates that the timezone attribute contains a valid value.
+     *
+     * @return void
+     * @throws TestReferenceException
+     */
+    private function validateTimezoneAttribute()
+    {
+        $attributes = $this->getCustomActionAttributes();
+        if (isset($attributes[self::ACTION_ATTRIBUTE_TIMEZONE])) {
+            $timezone = $attributes[self::ACTION_ATTRIBUTE_TIMEZONE];
+            if (array_search($timezone, timezone_identifiers_list()) === false) {
+                throw new TestReferenceException(
+                    "Timezone '{$timezone}' is not a valid timezone",
+                    ["stepKey" => $this->getStepKey(), self::ACTION_ATTRIBUTE_TIMEZONE => $timezone]
+                );
+            }
+        }
+    }
+
```

Comment:
No really an action item, just general observation that we seem to use different message format when throwing exception. e.g some provide stepKey and some don't, slightly different wording style, etc.